web/src/composebox_typeahead.js

```diff
@@ -1122,16 +1120,6 @@ function get_header_html() {
 }
 
 export function initialize_compose_typeahead(selector) {
-    const completions = {
-        mention: true,
-        emoji: true,
-        silent_mention: true,
-        slash: true,
-        stream: true,
-        syntax: true,
-        topic: true,
-        timestamp: true,
-    };
```

Comment:
So I think this actually should not be removed. The use case for it is that if we were to enable typeahead on the "stream description" input, or any other Markdown text input in the app, we would need to turn off a few features (like `slash` commands) in that typeahead, and this is the mechanism that we have to control that.

Reply:
It's not being used at all in the typeahead file though. If we don't remove it, we should make it a constant in this file. But that doesn't make that much sense to me given we're not currently using it for anything and can edit the code directly if we want.

Reply:
Making it a constant would not support what I suggest above.

We are doing something slightly evil here, which is to pass an extra option parameter for our own use, relying on the fact that the Bootstrap typeahead module will just happily include that in the typeahead object's `options` object.

I think that the right refactor is for `get_sorted_filtered_items` to accept a parameter that is `allowed_markdown_features` that is this object, and it can pass that on into `get_candidates`, such that we remove the evil part and make the data flow normal, without reindenting `get_candidates` in a way that we will likely need to revert later.

Reply:
Why can't we make `allowed_markdown_features` a constant local variable? I'm not clear on where you expect it to be mutated in the future and why we should pass it through to `get_candidates` like that instead of e.g. just defining it in `get_candidates` or directly above it.

Reply:
That's probably fine until we actually use the typeahead somewhere else.